a.el

```diff
@@ -124,10 +126,12 @@
     (length coll))
 
    ((hash-table-p coll)
-    (length (a-keys coll)))))
+    (hash-table-count coll))))
 
+;; 1) is it ok that collections of different types can be equal?
+;; 2) early termination would be nice for larger collections
 (defun a-equal (a b)
-  "Reduce an associative collection, starting with an initial value of FROM. The reducing functions receives the intermediate value, key, and value."
+  "Compare collections a and b for equality. Return true or false."
```

Comment:
Thanks, good catch! Probably should mention that it compares the set of key value pairs regardless of type. This one is also a bit of an odd one out in that it doesn't deal with vectors.